native build failed for smb+quartz

### Bug description

https://camel.zulipchat.com/#narrow/channel/257302-camel-quarkus/topic/native.20build.20failed.20for.20smb.2Bquartz/with/568173900

When project has dependencies for camel-quarkus-quartz&camel-quarkus-smb native build fails with stacktrace

```
Error: Class initialization of net.engio.mbassy.dispatch.el.StandardELResolutionContext failed. This error is reported at image build time because class net.engio.mbassy.dispatch.el.StandardELResolutionContext is registered for linking at image build time by command line and command line.
....................................
Caused by: java.lang.NoClassDefFoundError: javax/el/FunctionMapper
Caused by: java.lang.ClassNotFoundException: javax.el.FunctionMapper
```

**How to reproduce**
Clone https://github.com/apache/camel-quarkus-examples
Add dependencies to camel-quarkus-examples/timer-log/pom.xml
```
<dependency>
    <groupId>org.apache.camel.quarkus</groupId>
    <artifactId>camel-quarkus-quartz</artifactId>
</dependency>
<dependency>
    <groupId>org.apache.camel.quarkus</groupId>
    <artifactId>camel-quarkus-smb</artifactId>
</dependency>
```
Go to folder camel-quarkus-examples/timer-log and run command (build using Docker)
```
mvn -DskipTests=true package -Dnative -Dquarkus.native.container-build=true -Dquarkus.container-image.build=true -Dquarkus.container-image.push=false

```
I have checked all dependencies in project and didn't find mbassador dependency.
It seems, that mbassador used at build time only

And, if remove smb OR quartz from dependencies, each of them works fine separately.

--- a/extensions/smb/runtime/pom.xml
+++ b/extensions/smb/runtime/pom.xml
@@ -48,6 +48,15 @@
             <groupId>jakarta.el</groupId>
             <artifactId>jakarta.el-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>net.engio</groupId>
+            <artifactId>mbassador</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.graalvm.sdk</groupId>
+            <artifactId>graal-sdk</artifactId>
+            <scope>provided</scope>
+        </dependency>
     </dependencies>
 
     <build>

--- a/extensions/smb/runtime/src/main/java/org/apache/camel/component/smb/graal/ElFilterSubstitutions.java
+++ b/extensions/smb/runtime/src/main/java/org/apache/camel/component/smb/graal/ElFilterSubstitutions.java
@@ -1,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.smb.graal;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.RecomputeFieldValue;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+import jakarta.el.ExpressionFactory;
+import net.engio.mbassy.dispatch.el.StandardELResolutionContext;
+import net.engio.mbassy.subscription.SubscriptionContext;
+
+/**
+ * Disables unwanted for smbj jakarta.el support in net.engio:mbassador
+ *
+ * TODO: Remove this class when mbassador > 1.3.0 is available - https://github.com/apache/camel-quarkus/issues/5646
+ */
+public class ElFilterSubstitutions {
+    @TargetClass(className = "net.engio.mbassy.dispatch.el.ElFilter")
+    static final class SubstituteElFilter {
+        @Substitute
+        public static boolean isELAvailable() {
+            return false;
+        }
+
+        @Substitute
+        public static ExpressionFactory ELFactory() {
+            return null;
+        }
+
+        @Substitute
+        public boolean accepts(Object message, final SubscriptionContext context) {
+            return false;
+        }
+
+        @Substitute
+        private boolean evalExpression(final String expression,
+                final StandardELResolutionContext resolutionContext,
+                final SubscriptionContext context,
+                final Object message) {
+            return false;
+        }
+    }
+
+    @TargetClass(className = "net.engio.mbassy.dispatch.el.ElFilter$ExpressionFactoryHolder")
+    static final class SubstituteExpressionFactoryHolder {
+        @Alias
+        @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.FromAlias)
+        public static ExpressionFactory ELFactory = null;
+    }
+}

--- a/pom.xml
+++ b/pom.xml
@@ -143,6 +143,7 @@
         <langchain4j.version>1.5.0</langchain4j.version><!-- @sync io.quarkiverse.langchain4j:quarkus-langchain4j-parent:${quarkiverse-langchain4j.version} prop:langchain4j.version -->
         <langchain4j-beta.version>1.5.0-beta11</langchain4j-beta.version><!-- @sync io.quarkiverse.langchain4j:quarkus-langchain4j-parent:${quarkiverse-langchain4j.version} prop:langchain4j-beta.version -->
         <mapstruct.version>${mapstruct-version}</mapstruct.version>
+        <mbassador.version>1.3.0</mbassador.version><!-- @sync com.hierynomus:smbj:${smbj-version} dep:net.engio:mbassador -->
         <minio.version>8.6.0</minio.version><!-- @sync io.quarkiverse.minio:quarkus-minio-parent:${quarkiverse-minio.version} prop:minio.version -->
         <msal4j.version>1.23.1</msal4j.version><!-- @sync com.azure:azure-identity:${azure-identity.version} dep:com.microsoft.azure:msal4j -->
         <murmur.version>1.0.0</murmur.version>

--- a/poms/bom/pom.xml
+++ b/poms/bom/pom.xml
@@ -7810,6 +7810,11 @@
                 <version>${jodatime.version}</version>
             </dependency>
             <dependency>
+                <groupId>net.engio</groupId>
+                <artifactId>mbassador</artifactId>
+                <version>${mbassador.version}</version>
+            </dependency>
+            <dependency>
                 <groupId>net.i2p.crypto</groupId>
                 <artifactId>eddsa</artifactId>
                 <version>${eddsa.version}</version>

--- a/poms/bom/src/main/generated/flattened-full-pom.xml
+++ b/poms/bom/src/main/generated/flattened-full-pom.xml
@@ -7696,6 +7696,11 @@
         <version>2.14.0</version><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->
       </dependency>
       <dependency>
+        <groupId>net.engio</groupId><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->
+        <artifactId>mbassador</artifactId><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->
+        <version>1.3.0</version><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->
+      </dependency>
+      <dependency>
         <groupId>net.i2p.crypto</groupId><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->
         <artifactId>eddsa</artifactId><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->
         <version>0.3.0</version><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->

--- a/poms/bom/src/main/generated/flattened-reduced-pom.xml
+++ b/poms/bom/src/main/generated/flattened-reduced-pom.xml
@@ -7640,6 +7640,11 @@
         <version>2.14.0</version>
       </dependency>
       <dependency>
+        <groupId>net.engio</groupId>
+        <artifactId>mbassador</artifactId>
+        <version>1.3.0</version>
+      </dependency>
+      <dependency>
         <groupId>net.i2p.crypto</groupId>
         <artifactId>eddsa</artifactId>
         <version>0.3.0</version>

--- a/poms/bom/src/main/generated/flattened-reduced-verbose-pom.xml
+++ b/poms/bom/src/main/generated/flattened-reduced-verbose-pom.xml
@@ -7640,6 +7640,11 @@
         <version>2.14.0</version><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->
       </dependency>
       <dependency>
+        <groupId>net.engio</groupId><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->
+        <artifactId>mbassador</artifactId><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->
+        <version>1.3.0</version><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->
+      </dependency>
+      <dependency>
         <groupId>net.i2p.crypto</groupId><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->
         <artifactId>eddsa</artifactId><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->
         <version>0.3.0</version><!-- org.apache.camel.quarkus:camel-quarkus-bom:${project.version} -->

